Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -379,6 +408,14 @@ void MainWindow::ShowAboutDialog()
   about->show();
 }
 
+void MainWindow::ShowHotkeyDialog()
+{
+  m_hotkey_window->ChangeMappingType(MappingWindow::Type::MAPPING_HOTKEYS);
+  m_hotkey_window->show();
```

Comment:
Not `exec()`? Then you could do away with storing the pointer in `MainWindow` and create and destroy the dialog entirely in this method.